tests/Dfc.CourseDirectory.WebV2.Tests/FeatureTests/DataManagement/Venues/UploadTests.cs

```diff
@@ -57,14 +115,89 @@ public async Task Post_ValidVenuesFile_CreatesRecordAndRedirectsToPublish()
 
             // Assert
             response.StatusCode.Should().Be(HttpStatusCode.Redirect);
-            response.Headers.Location.Should().Be($"/data-upload/venues/check-publish?providerId={provider.ProviderId}");
+            response.Headers.Location.Should().Be($"/data-upload/venues/in-progress?providerId={provider.ProviderId}");
 
             SqlQuerySpy.VerifyQuery<CreateVenueUpload, Success>(q =>
                 q.CreatedBy.UserId == User.UserId &&
                 q.CreatedOn == Clock.UtcNow &&
                 q.ProviderId == provider.ProviderId);
         }
 
+        [Fact]
+        public async Task Post_ValidUpload_AbandonsExistingUnpublishedUpload()
+        {
+            // Arrange
+            var provider = await TestData.CreateProvider();
+
+            var oldUpload = await TestData.CreateVenueUpload(provider.ProviderId, createdBy: User.ToUserInfo(), UploadStatus.Processed);
+
+            var row1 = new VenueRow()
+            {
+                AddressLine1 = Faker.Address.StreetAddress(),
+                Postcode = Faker.Address.UkPostCode(),
+                VenueName = Faker.Company.Name()
+            };
+
+            var csvStream = CreateCsvStream(new[] { row1 });
+            var requestContent = CreateMultiPartDataContent("text/csv", csvStream);
+
+            // Act
+            var response = await HttpClient.PostAsync($"/data-upload/venues/upload?providerId={provider.ProviderId}", requestContent);
+
+            // Assert
+            response.EnsureNonErrorStatusCode();
+
+            oldUpload = await WithSqlQueryDispatcher(
+                dispatcher => dispatcher.ExecuteQuery(new GetVenueUpload() { VenueUploadId = oldUpload.VenueUploadId }));
+            oldUpload.UploadStatus.Should().Be(UploadStatus.Abandoned);
+        }
+
+        [Fact(Skip = "Flaky on CI")]
```

Comment:
Temporary until we get a way of 'spying' on `IVenueUploadProcessor` in tests

Reply:
https://github.com/SkillsFundingAgency/dfc-coursedirectory/pull/2021 will fix this